pylib/Stages/Reporter/JunitXML.py

```diff
@@ -57,7 +57,7 @@ def execute(self, log, keyvals, testDef):
         cmds = {}
         testDef.parseOptions(log, self.options, keyvals, cmds)
         if cmds['filename'] is not None:
-            self.fh = open(cmds['filename'], 'w')
+            self.fh = open(os.path.join(cmds['scratch'],cmds['filename']), 'w')
```

Comment:
Would it be worth it to check if `cmds['filename']` is an absolute path before appending it to the scratch directory?

Reply:
I am assuming that cmds['scratch'] is already an absolute path when that is set by TestDef

Reply:
Right, but if the user sets the reporter's file to `/home/user/not/in/scratch/junit.xml` and scratch is at `/home/user/scratch`, this will output it to `/home/user/scratch/home/user/not/in/scratch/junit.xml`. So I am asking if it is worth it to check  if the filename is already an absolute path.

If it should _just_ be a filename and not a path to a file then it should be fine left as-is IMO. Just wanted your input there.

Reply:
I see

I will add a check for abs path before joining the paths then